code/modules/reagents/chemistry/holder.dm

```diff
@@ -630,7 +630,7 @@
 	var/list/cached_reagents = reagent_list
 	for(var/_reagent in cached_reagents)
 		var/datum/reagent/R = _reagent
-		if (R.id == reagent)
+		if (R.id == reagent && R.volume > 0)
 			return R.volume
```

Comment:
`return max(R.volume, 0)` instead, that way the loop ends when you've found the reagent instead of the loop continuing to the end even though you already found the reagent and intentionally skipped it over